client/signup/steps/user/index.jsx

```diff
@@ -208,20 +208,22 @@ export class UserStep extends Component {
 
 		this.props.recordTracksEvent( 'calypso_signup_user_step_submit', analyticsData );
 
-		const shouldRecordRecaptchaAction =
-			typeof this.state.recaptchaClientId === 'number' &&
-			'onboarding' === this.props.flowName &&
-			'show' === abtest( 'userStepRecaptcha' );
+		const isRecaptchaLoaded = typeof this.state.recaptchaClientId === 'number';
+		const isRecaptchaABTest =
+			'onboarding' === this.props.flowName && 'show' === abtest( 'userStepRecaptcha' );
 
-		const recaptchaPromise = shouldRecordRecaptchaAction
-			? recordGoogleRecaptchaAction( this.state.recaptchaClientId, 'calypso/signup/formSubmit' )
-			: Promise.resolve();
+		const recaptchaPromise =
+			isRecaptchaABTest && isRecaptchaLoaded
+				? recordGoogleRecaptchaAction( this.state.recaptchaClientId, 'calypso/signup/formSubmit' )
+				: Promise.resolve();
```

Comment:
Another option is to turn the `submitForm` method to async function and then `await` conditionally:
```js
if ( isRecaptchaABTest && isRecaptchaLoaded ) {
  await recordGoogleRecaptchaAction( ... )
}

this.submit( ... );
```

Another code style nit: you're checking `isRecaptchaABTest` too often, making the code logic obfuscated. What about:
```js
// default values
let recaptchaDidntLoad = false;
let recaptchaFailed = false;

if ( isRecaptchaABTest ) {
  // do all the recaptcha work inside this block
}

// use the values
this.submit( {
  recaptchaDidntLoad,
  recaptchaFailed
} );
```